packages/browser/src/login/oidc/redirectHandler/AuthCodeRedirectHandler.ts

```diff
@@ -106,8 +105,9 @@ export class AuthCodeRedirectHandler implements IRedirectHandler {
         `AuthCodeRedirectHandler cannot handle [${redirectUrl}]: it is missing one of [code, state].`
       );
     }
-    const url = new URLParse(redirectUrl, true);
-    const oauthState = url.query.state as string;
+
+    const url = new URL(redirectUrl);
+    const oauthState = url.searchParams.get("state") as string;
```

Comment:
Here, as in `OidcLoginHandler`, I'd rather use proper type guards to make the check for the property explicit, rather than use an equivalent type assertion.

Reply:
Not possible to add type guards on a Map lookup.